internal/olm/operator/registry/configmap/configmap_test.go

```diff
@@ -0,0 +1,277 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	client_cr "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/yaml"
+)
+
+var _ = Describe("ConfigMap", func() {
+
+	Describe("hashContents", func() {
+		It("should return the hash of the byte array without any error", func() {
+			val := []byte("Hello")
+			h := sha256.New()
+			_, _ = h.Write(val)
+			enc := base32.StdEncoding.WithPadding(base32.NoPadding)
+			ans := enc.EncodeToString(h.Sum(nil))
+
+			Expect(hashContents([]byte("Hello"))).Should(Equal(ans))
+		})
+	})
+
+	Describe("getRegistryConfigMapName", func() {
+		It("should return the registry configmap name", func() {
+			val := "Test"
+			name := k8sutil.FormatOperatorNameDNS1123(val)
+			ans := fmt.Sprintf("%s-registry-manifests", name)
+
+			Expect(getRegistryConfigMapName("Test")).Should(Equal(ans))
+		})
+	})
+
+	Describe("makeObjectFileName", func() {
+		var (
+			fileName  string
+			testVal   []byte
+			userInput []string
+		)
+		BeforeEach(func() {
+			testVal = []byte("test")
+			userInput = []string{"userInput", "userInput2"}
+			fileName = hashContents(testVal) + "."
+		})
+
+		It("returns the filename with extra user given string", func() {
+			for _, name := range userInput {
+				if name != "" {
+					fileName += strings.ToLower(name) + "."
+				}
+			}
+			fileName = fileName + "yaml"
+			Expect(makeObjectFileName(testVal, userInput...)).Should(Equal(fileName))
+		})
+
+		It("returns the filename without extra user given string", func() {
+			fileName = fileName + "yaml"
+			Expect(makeObjectFileName(testVal)).Should(Equal(fileName))
+		})
+	})
+
+	Describe("addObjectToBinaryData ", func() {
+		It("should add given object to the given binaryData", func() {
+			type binaryData map[string][]byte
+			userInput := []string{"userInput", "userInput2"}
+
+			type test_struct struct {
+				val1 string
+				val2 string
+			}
+			b := make(binaryData)
+			// fmt.Printf("%+v", b)
+			obj := test_struct{"val1", "val2"}
+
+			binaryData_local := make(binaryData)
+			b_local, err := yaml.Marshal(obj)
+
+			binaryData_local[makeObjectFileName(b_local, userInput...)] = b_local
+
+			Expect(err).Should(BeNil())
+			Expect(addObjectToBinaryData(b, obj, userInput...)).Should(BeNil())
+			Expect(b).Should(Equal(binaryData_local))
+		})
+
+	})
+
+	Describe("makeObjectBinaryData", func() {
+		It("creates the binary data", func() {
+			binaryData := make(map[string][]byte)
+			type return_struct struct {
+				val1 map[string][]byte
+				val2 error
+			}
+			type test_struct struct {
+				val1 string
+				val2 string
+			}
+
+			obj := test_struct{"val1", "val2"}
+			userInput := []string{"userInput", "userInput2"}
+			b := make(map[string][]byte)
+			b, e := makeObjectBinaryData(obj, userInput...)
+
+			addObjectToBinaryData(binaryData, obj, userInput...)
+
+			Expect(e).Should(BeNil())
```

Comment:
this check for e should be below the `makeObjectBinaryData` call since it isn't a test expectation.